File: qh_241/sinh.cpp
#include <bits/stdc++.h>

using namespace std;

mt19937_64 rng(chrono::high_resolution_clock::now().time_since_epoch().count());
int randint(int l, int r) { return uniform_int_distribution<int>(l, r)(rng); }

signed main() {
  srand(time(0));

  system("g++ chuan.cpp -o chuan");
  system("g++ trau.cpp -o trau");

  int limTest = 100;
  int lim = 1e5;
  for (int test = 1; test <= limTest; test++) {
    ofstream inp("test.inp");

    int n = randint(1, lim);
    inp << n << "\n";
    for (int i = 0; i < n; i++) {
      inp << randint(1, 100) << " ";
    }

    inp.close();

    cout << "--- RUNNING ON TEST " << test << " ---\n";
    cout << "RESULT: ";
    system("./chuan");
    system("./trau");

    int check = system("diff chuan.out trau.out");
    if (check) {
      cout << "WA!!!\n\n";
      return 0;
    }
    cout << "AC\n\n";
  }

  cout << "ALL TEST PASSED SUCCESSFULL\n";
}

File: qh_241/sinh.py
from subprocess import run, TimeoutExpired, CalledProcessError
from random import randint

LIM = 50000
TEST = 100
CHUAN = "chuan"
TRAU = "trau"


def gen(lim):
    n = randint(1, lim)
    a = []
    for i in range(n):
        a.append(str(randint(1, lim)))

    with open("test.inp", "w") as f:
        f.write(f"{n}\n")
        f.write(f"{" ".join(a)}")


def exc(prog, time):
    try:
        run([f"./{prog}"], timeout=time, check=1)
    except TimeoutExpired:
        print(f"[TLE] - {prog}")
        exit()
    except CalledProcessError as e:
        print(f"[RE]: {prog} | code: {e}")
        exit()


def comp():
    run(["g++", f"{CHUAN}.cpp", "-o", f"{CHUAN}"])
    run(["g++", f"{TRAU}.cpp", "-o", f"{TRAU}"])
    for test in range(TEST):
        gen(LIM)

        print(f"TEST {test + 1}: ", end="")

        exc(CHUAN, 1)
        exc(TRAU, 1)

        with open(f"{CHUAN}.out", "r") as f:
            chuan = f.read()

        with open(f"{TRAU}.out", "r") as f:
            trau = f.read()

        if chuan != trau:
            print("[WA]")
            exit()
        else:
            print("[AC]")

    print("ALL TEST PASSED")


comp()
